pkg/worker/k8srunner.go

```diff
@@ -226,25 +243,43 @@ func (r k8sStepRunner) RunStep(ctx context.Context) StepResult {
 	}
 }
 
-func (r k8sStepRunner) dryRunStepError(ctx context.Context) error {
+func (r k8sStepRunner) runStep(ctx context.Context) error {
+	if r.DryRun != DryRunNone {
+		return r.dryRunStep(ctx)
+	}
+	return r.liveRunStep(ctx)
+}
+
+func (r k8sStepRunner) dryRunStep(ctx context.Context) error {
+	if r.DryRun == DryRunClient {
```

Comment:
Can't --dry-run=client be forwarded to the actual kubectl command? Just trying to learn here :)

Reply:
We're not running kubectl under the hood. Only the Kubernetes api-client, which kubectl also uses.

IIRC what `kubectl --dry-run` does is parse the YAML/JSON manifests you give it and compare that with some JSON schemas.

We're not creating any objects from YAML. We're creating the objects directly in memory. So we don't need to validate them.